From e21f33695da3275ec81b5f94685f0e4ac92c9ad5 Mon Sep 17 00:00:00 2001
From: Gyeongjae Choi <def6488@gmail.com>
Date: Mon, 30 Oct 2023 14:35:04 +0000
Subject: [PATCH 17/17] Remove test modules that fail to build

These are tests and they have both void vs int return value problems and implicit
function argument cast problems. Not worth fixing for tests.

---
 scipy/integrate/meson.build | 18 ------------------
 scipy/io/meson.build        | 21 ---------------------
 2 files changed, 39 deletions(-)

diff --git a/scipy/integrate/meson.build b/scipy/integrate/meson.build
index ae9e2466e1..e11626db0d 100644
--- a/scipy/integrate/meson.build
+++ b/scipy/integrate/meson.build
@@ -187,24 +187,6 @@ py3.extension_module('_test_multivariate',
   install_tag: 'tests'
 )
 
-_test_odeint_banded_module = custom_target('_test_odeint_banded_module',
-  output: ['_test_odeint_bandedmodule.c', '_test_odeint_banded-f2pywrappers.f'],
-  input: 'tests/test_odeint_banded.pyf',
-  command: [generate_f2pymod, '@INPUT@', '-o', '@OUTDIR@']
-)
-
-py3.extension_module('_test_odeint_banded',
-  ['tests/banded5x5.f', _test_odeint_banded_module],
-  link_with: [lsoda_lib, mach_lib],
-  fortran_args: _fflag_Wno_unused_dummy_argument,
-  link_args: version_link_args,
-  dependencies: [lapack_dep, fortranobject_dep],
-  install: true,
-  link_language: 'fortran',
-  subdir: 'scipy/integrate',
-  install_tag: 'tests'
-)
-
 subdir('_ivp')
 subdir('tests')
 
diff --git a/scipy/io/meson.build b/scipy/io/meson.build
index d6fc6dc749..af04022208 100644
--- a/scipy/io/meson.build
+++ b/scipy/io/meson.build
@@ -1,24 +1,3 @@
-_test_fortran_module = custom_target('_test_fortran_module',
-  output: ['_test_fortranmodule.c'],
-  input: 'test_fortran.pyf',
-  command: [generate_f2pymod, '@INPUT@', '-o', '@OUTDIR@']
-)
-
-py3.extension_module('_test_fortran',
-  [
-    _test_fortran_module,
-    '_test_fortran.f'
-  ],
-  c_args: [Wno_unused_variable],
-  fortran_args: fortran_ignore_warnings,
-  link_args: version_link_args,
-  dependencies: [lapack_dep, fortranobject_dep],
-  install: true,
-  link_language: 'fortran',
-  subdir: 'scipy/io',
-  install_tag: 'tests'
-)
-
 py3.install_sources([
     '__init__.py',
     '_fortran.py',
-- 
2.39.3 (Apple Git-146)

